sql/catalyst/src/main/scala/org/apache/spark/sql/internal/connector/PredicateUtils.scala

```diff
@@ -132,4 +133,18 @@ private[sql] object PredicateUtils {
       case _ => None
     }
   }
+
+  def toV1(
+      predicates: Array[Predicate],
+      skipIfNotConvertible: Boolean): Array[Filter] = {
+    predicates.flatMap { predicate =>
+      val filter = toV1(predicate)
+      if (filter.isEmpty) {
+        if (!skipIfNotConvertible) {
```

Comment:
should we merge these two if as `if (filter.isEmpty && !skipIfNotConvertible)`

Reply:
Fixed. Thanks